xapian-core/tests/api_cluster.cc

```diff
@@ -51,6 +51,58 @@ make_stemmed_cluster_db(Xapian::WritableDatabase &db, const std::string &)
     }
 }
 
+/** Round Robin clusterer:
+ *  This clusterer is a minimal clusterer which will cluster documents as -
+ *  ith document goes to the (i % k)th cluster where k is the number of clusters and
+ *  0 <= i < N; where N is the number of documents
+ */
+class XAPIAN_VISIBILITY_DEFAULT RoundRobin : public Xapian::Clusterer {
```

Comment:
Should I rebase and squash?
Also, even after the stemming branch has been merged, the commits diff doesn't seem to change. It looks like commits from the stemming branch are still here. Is there any way to remove them to have a cleaner history on this PR? I was of the idea that the commit diff reduces when the same commits are added to the target branch.